### PR TITLE
Issue 1850: Streams and scopes system test

### LINKED ISSUE #1850: System test to exercise creation, deletion, and sealing of scopes and streams
**Problem description**
We need to exercise regularly the creation, deletion, and sealing of scopes and streams. 

**Problem location**
Tests.

**Suggestions for an improvement**
Add a system test to do it.

### PR DESCRIPTION
**Pull request by RaulGracia**
**Change log description**
Creation of a new system test specifically addressed to check management operations on streams and scopes (`StreamsAndScopesManagementTest.java`).

**Purpose of the change**
Fixes #1850.

**What the code does**
The test executes a series of metadata operations on streams and scopes to verify their correct operation. This includes:
* The creation and deletion of multiple scopes both in correct and correct situations. 
* For each scope, the test creates a range of streams and tries to create, update, seal and delete them in correct and incorrect situations. 
* The test performs metadata operations on empty and non-empty streams.
* The test also provides performance metrics on metadata operations.

Note: the test is prepared to check re-creation of streams (`TEST_ITERATION > 1`) when issues #1012 and #1681 are solved. 

**How to verify it**
Execute the new system test in cluster mode.

### CODE REVIEW

**Comment by RaulGracia** [comment]
[System test to exercise creation, deletion, and sealing of scopes and streams](https://trello.com/c/YJNsoIfT/44-system-test-to-exercise-creation-deletion-and-sealing-of-scopes-and-streams)

File: test/system/src/test/java/io/pravega/test/system/StreamsAndScopesManagementTest.java [hunk 1]
```diff
@@ -0,0 +1,254 @@
...
+import static org.junit.Assert.assertTrue;
+
+@Slf4j
+@RunWith(SystemTestRunner.class)
+public class StreamsAndScopesManagementTest {
+
+    private static final int NUM_SCOPES = 5;
+    private static final int NUM_STREAMS = 20;
+    private static final int NUM_EVENTS = 100;
+    // TODO: Re-creation of Streams cannot be tested (Issue #2641).
```
Comment (Prabha-Veerubhotla) [inline test/system/src/test/java/io/pravega/test/system/StreamsAndScopesManagementTest.java]: make it  `TODO:  https://github.com/pravega/pravega/issues/2641`

File: test/system/src/test/java/io/pravega/test/system/StreamsAndScopesManagementTest.java [hunk 2]
```diff
@@ -0,0 +1,254 @@
...
+    private static final int TEST_ITERATIONS = 1;
+    @Rule
+    public Timeout globalTimeout = Timeout.seconds(8 * 60);
+
+    private final ScheduledExecutorService executor = ExecutorServiceHelpers.newScheduledThreadPool(4, "executor");
+
+    private URI controllerURI = null;
+    private StreamManager streamManager = null;
+    private Controller controller;
+
```
Comment (Prabha-Veerubhotla) [inline test/system/src/test/java/io/pravega/test/system/StreamsAndScopesManagementTest.java]: remove extra line.

File: test/system/src/test/java/io/pravega/test/system/StreamsAndScopesManagementTest.java [hunk 3]
```diff
@@ -0,0 +1,254 @@
...
+    }
+
+    @Before
+    public void setup() {
+        Service conService = Utils.createPravegaControllerService(null);
+        List<URI> ctlURIs = conService.getServiceDetails();
+        controllerURI = ctlURIs.get(0);
+        streamManager = StreamManager.create(controllerURI);
+        controller = new ControllerImpl(ControllerImplConfig.builder()
+                                                            .clientConfig(ClientConfig.builder().controllerURI(controllerURI).build())
```
Comment (Prabha-Veerubhotla) [inline test/system/src/test/java/io/pravega/test/system/StreamsAndScopesManagementTest.java]: indentation seems incorrect. Please fix it.

File: test/system/src/test/java/io/pravega/test/system/StreamsAndScopesManagementTest.java [hunk 4]
```diff
@@ -0,0 +1,254 @@
...
+
+    private static final int NUM_SCOPES = 5;
+    private static final int NUM_STREAMS = 20;
+    private static final int NUM_EVENTS = 100;
+    // TODO: Re-creation of Streams cannot be tested (Issue #2641).
+    private static final int TEST_ITERATIONS = 1;
+    @Rule
+    public Timeout globalTimeout = Timeout.seconds(8 * 60);
+
+    private final ScheduledExecutorService executor = ExecutorServiceHelpers.newScheduledThreadPool(4, "executor");
```
Comment (Prabha-Veerubhotla) [inline test/system/src/test/java/io/pravega/test/system/StreamsAndScopesManagementTest.java]: you can make it 
` ExecutorServiceHelpers.newScheduledThreadPool(4, " StreamsAndScopesManagementTest-controller");`

File: test/system/src/test/java/io/pravega/test/system/StreamsAndScopesManagementTest.java (hunk 1 above)
Comment (RaulGracia) [reply test/system/src/test/java/io/pravega/test/system/StreamsAndScopesManagementTest.java]: Done.

File: test/system/src/test/java/io/pravega/test/system/StreamsAndScopesManagementTest.java (hunk 2 above)
Comment (RaulGracia) [reply test/system/src/test/java/io/pravega/test/system/StreamsAndScopesManagementTest.java]: Done.

File: test/system/src/test/java/io/pravega/test/system/StreamsAndScopesManagementTest.java (hunk 3 above)
Comment (RaulGracia) [reply test/system/src/test/java/io/pravega/test/system/StreamsAndScopesManagementTest.java]: Done.

File: test/system/src/test/java/io/pravega/test/system/StreamsAndScopesManagementTest.java (hunk 4 above)
Comment (RaulGracia) [reply test/system/src/test/java/io/pravega/test/system/StreamsAndScopesManagementTest.java]: Done.

File: test/system/src/test/java/io/pravega/test/system/StreamsAndScopesManagementTest.java [hunk 5]
```diff
@@ -0,0 +1,254 @@
...
+        assertTrue("Deleting scope", streamManager.deleteScope(scope));
+        controllerPerfStats.get("deleteScopeMs").add(timeDiffInMs(iniTime));
+    }
+
+    private void testCreateSealAndDeleteStreams(String scope) {
+        for (int j = 1; j <= NUM_STREAMS; j++) {
+            final String stream = String.valueOf(j);
+            StreamConfiguration config = StreamConfiguration.builder().scalingPolicy(ScalingPolicy.fixed(j)).build();
+
+            // Create Stream with wrong scope, which should not be successful.
```
Comment (Prabha-Veerubhotla) [inline test/system/src/test/java/io/pravega/test/system/StreamsAndScopesManagementTest.java]: You can modify the `wrong scope` to `nonexistent scope.`

Comment (RaulGracia) [reply test/system/src/test/java/io/pravega/test/system/StreamsAndScopesManagementTest.java]: Done.

**Pull request merged**